app/src/main/java/com/coveros/coverosmobileapp/oauth/RestRequest.java

```diff
@@ -151,126 +141,86 @@ public String getBodyContentType() {
+     *
+     * @param onAuthFailedListener
      */
-    private boolean isJsonBoolean(String jsonString) {
-        try {
-            JSONObject jsonObject = new JSONObject();
-            if (jsonString.equals(Boolean.TRUE.toString())) {
-                jsonObject.put(ORIGINAL_RESPONSE_TAG, true);
-            }
-            if (jsonString.equals(Boolean.FALSE.toString())) {
-                jsonObject.put(ORIGINAL_RESPONSE_TAG, false);
-            }
-            return true;
-        } catch (JSONException je) {
-            Log.d(APP_NAME, "isJsonBoolean is false", je);
-            return false;
+    public void setOnAuthFailedListener(OnAuthFailedListener onAuthFailedListener) {
+        if (isAuthenticated) {
+            this.onAuthFailedListener = onAuthFailedListener;
+        } else {
+            Log.e("RestRequest", "Not an authenticated request: onAuthFailedListener was not set");
         }
     }
 
-    /**
-     * Creates JSONObject from a string representation of a JSON.
-     * @param jsonString    string to be used to create JSONObject
-     * @return JSONObject    JSONObject created from jsonString
-     * @throws JSONException    should never be thrown if used in conjunction with isJson[]() methods above
-     */
-    protected JSONObject jsonObjectFromResponse(String jsonString) throws JSONException {
-        return new JSONObject(jsonString);
-    }
-
-    /**
-     * Creates JSONObject from a string representation of a JSON array.
-     * @param jsonString    string to be used to create JSONObject
-     * @return wrapper    JSONObject created from jsonString
-     * @throws JSONException    should never be thrown if used in conjunction with isJson[]() methods above
-     */
-    protected JSONObject jsonArrayObjectFromResponse(String jsonString) throws JSONException {
-        JSONArray responseArray = new JSONArray(jsonString);
-        JSONObject wrapper = new JSONObject();
-        wrapper.put(ORIGINAL_RESPONSE_TAG, responseArray);
-
-        return wrapper;
-    }
-
-    /**
-     * Creates JSONObject from a string representation of a JSON boolean.
-     * @param jsonString    string to be used to create JSONObject
-     * @return jsonObject    JSONObject created from jsonString
-     * @throws JSONException    should never be thrown if used in conjunction with isJson[]() methods above
-     */
-    protected JSONObject jsonBooleanObjectFromResponse(String jsonString) throws JSONException {
-        JSONObject jsonObject = new JSONObject();
-
-        if (jsonString.equals(Boolean.TRUE.toString())) {
-            jsonObject.put(ORIGINAL_RESPONSE_TAG, true);
-            return jsonObject;
-        } else if (jsonString.equals(Boolean.FALSE.toString())) {
-            jsonObject.put(ORIGINAL_RESPONSE_TAG, false);
-            return jsonObject;
-        } else {
-            throw new JSONException("Not a valid JSON response: " + jsonString);
-        }
+    public boolean getIsAuthenticated() {
```

Comment:
The naming convention for boolean getters is "isField", so in this case, your method should be "isAuthenticated".